test/unit/ro_test.lua

```diff
@@ -0,0 +1,131 @@
+local expirationd = require("expirationd")
+local fiber = require("fiber")
+local t = require("luatest")
+local helpers = require("test.helper")
+local g = t.group('expirationd_ro')
+
+g.before_all(function()
+    -- we need to restore a state after fail
+    g.default_box_cfg = {read_only = box.cfg.read_only}
+end)
+
+g.before_each(function()
+    box.cfg({read_only = false})
+    g.space = helpers.create_space_with_tree_index('memtx')
+    box.cfg(g.default_box_cfg)
+end)
+
+g.after_each(function()
+    box.cfg({read_only = false})
+    g.space:drop()
+    box.cfg(g.default_box_cfg)
+end)
+
+local function create_id_index(space)
+    space:create_index("primary", {
+        type = "TREE",
+        parts = {
+            {
+                field = 1
+            }
+        }
+    })
+end
+
+local tuples = {{1, "1"}, {2, "2"}, {3, "3"}}
+local function insert_tuples(space)
+    for i = 1,3 do
+        space:insert({i, tostring(i)})
+    end
+end
+
+function g.test_ro_temporary()
+    box.cfg({read_only = false})
+    local space = box.schema.create_space("temporary", {temporary = true})
+    create_id_index(space)
+    insert_tuples(space)
+
+    box.cfg({read_only = true})
+
+    helpers.iteration_result = {}
+    local task = expirationd.start("clean_all", space.id, helpers.is_expired_debug,
+            {full_scan_delay = 0})
+
+    helpers.retrying({}, function()
+        t.assert_equals(helpers.iteration_result, tuples)
+    end)
+
+    task:kill()
+    box.cfg({read_only = false})
+    space:drop()
```

Comment:
If assert fails, space wouldn't be dropped.

Reply:
On the other hand. it doesn't seems like it should break anything.

Reply:
Thank you, fixed.